### PR TITLE
Query join support

### PR DESCRIPTION
**Pull request by u007**
added query join support
updated docs to show sample find, query and join

### CODE REVIEW

**Comment by u007** [comment]
realigned function ordering back to the original, please check again if you have checked before

**Comment by markbates** [comment]
It looks like `joinClause.Arguments` are never used. If so, can we remove them to make the syntax a bit nicer?

**Comment by markbates** [comment]
I'm also wondering if we can remove the `as` argument and let people pass it in as the table name. We can extract it from there. I'm just trying to see if we can get the syntax a bit cleaner. 

If we could make this:

```go
query = pop.Q(tx).Where("id = ?", 1).Join("books", "", []string{"1=1", "user_id=?"}, "xx").Order("name asc")
```

into this:

```go
query = pop.Q(tx).Where("id = ?", 1).Join("books as b", "1=1", "user_id=?").Order("name asc")
```

that would rock!

**Comment by markbates** [comment]
Also, let's split the join functions out into their own file `joins.go` just to make it a little easier to find.

Great work on this!

**Comment by u007** [comment]
@markbates joinClause.Arguments may be in use as parameters from input, sometimes its possible to need join condition with parameters from input.

As is required, because there are time that you may need to join the same table twice to get parent row, and child row recursively on the same table.
Example:
from users left join users parent_user on users.parent_id=parent_user.id
left join users child_user on users.id=child_user.parent_id

yes, your proposed "books as b" sounds good, will implement soon...
okay, noted on the flexible arguments as conditions

**Comment by markbates** [comment]
Ok, I understand about the `Arguments` now. That makes sense. In that case why not do what `Where` does? 

```go
func (q *Query) Where(stmt string, args ...interface{}) *Query
```

So we could make it this:

```go
query = pop.Q(tx).Where("id = ?", 1).Join("books as b", "1=1 and user_id=?", "xx").Order("name asc")
```

**Comment by u007** [comment]
you are right, arguments can be a where clause. btw, we are also missing group by right?

**Comment by u007** [comment]
Hi @markbates, is everything okay now?

**Comment by Michsior14** [comment]
@u007 I think @markbates meant that you should change on argument from `[]string` to `string` e.g 
`func (q *Query) Join(table string, on string, args ...interface{}) *Query` 
so users can specify statement on theirs own and simplify the markup.

**Comment by markbates** [comment]
Sorry, I forgot about this PR, but what @Michsior14 said is correct.

File: README.md [hunk 1]
```diff
@@ -220,3 +220,36 @@ Migrations can also be run in reverse to rollback the schema.
...
+err := query.All(&users)
+```
+
+##### Join Query
+
+```
+// page: page number
+// perpage: limit
+roles := []models.UserRole{}
+query := pop.Q(models.DB).LeftJoin("roles", "", []string{"roles.id=user_roles.role_id"}).
```
Comment (markbates) [inline README.md]: It looks like the docs are now out of date from the new implementation of `Join`. Can you please update? Thanks.

**Comment by markbates** [comment]
@u007 once the README is updated, this looks good to go! Awesome job!

**Comment by u007** [comment]
done

**Comment by markbates** [comment]
Woot! Awesome work! Thanks.

**Pull request merged**